docs/cvl/methods.md

```diff
@@ -313,6 +435,21 @@ variables defined as arguments in the summary declarations; expressions
 that combine those variables are not supported.
 
 There are a few restrictions on the functions that can be used as approximations:
+
  - Functions used as summaries are not allowed to call contract functions.
- - Functions used as summaries may not have accept arguments or return values that have struct or array types.
+
+ - The types of any arguments passed to or values returned from the summary
+   must be {ref}`convertible <type-conversions>` between CVL and Solidity types.
+   Arguments that are not accessed in the summary may have any type.
+
+Function summaries for *internal* methods have a few additional restrictions on
+their arguments and return types:
+ - arrays (including static arrays, `bytes`, and `string`) are not supported
+ - struct fields must have [value types][solidity-value-types]
+ - `storage` and `calldata` structs are not supported, only `memory`
+
+You can still summarize functions that take unconvertible types as arguments,
+but you cannot access those arguments in your summary.
```

Comment:
Do we need to explain anything about functions defined in libraries?

Reply:
I don't think so - I don't think there's anything special about them here.  They are already mentioned in the section that describes visibility modifiers.